Modules/Microsoft365DSC/DSCResources/MSFT_PPTenantSettings/MSFT_PPTenantSettings.psm1

```diff
@@ -0,0 +1,490 @@
+      DisableNewsletterSendout                       = $settings.disableNewsletterSendout
+      DisableEnvironmentCreationByNonAdminUsers      = $settings.disableEnvironmentCreationByNonAdminUsers
+      DisablePortalsCreationByNonAdminUsers          = $settings.disablePortalsCreationByNonAdminUsers
+      DisableSurveyFeedback                          = $settings.disableSurveyFeedback
+      DisableTrialEnvironmentCreationByNonAdminUsers = $settings.disableTrialEnvironmentCreationByNonAdminUsers
+      DisableCapacityAllocationByEnvironmentAdmins   = $settings.disableCapacityAllocationByEnvironmentAdmins
+      DisableSupportTicketsVisibleByAllUsers         = $settings.disableSupportTicketsVisibleByAllUsers
+      DisableDocsSearch                              = $settings.powerPlatform.search.disableDocsSearch
+      DisableCommunitySearch                         = $settings.powerPlatform.search.disableCommunitySearch
+      DisableBingVideoSearch                         = $settings.powerPlatform.search.disableBingVideoSearch
+      DisableShareWithEveryone                       = $settings.powerPlatform.powerApps.disableShareWithEveryone
+      EnableGuestsToMake                             = $settings.powerPlatform.powerApps.enableGuestsToMake
+      ShareWithColleaguesUserLimit                   = $settings.powerPlatform.teamsIntegration.shareWithColleaguesUserLimit
+      GlobalAdminAccount                             = $GlobalAdminAccount
+    }
+    $Results = Get-TargetResource @Params
+    $Results = Update-M365DSCExportAuthenticationResults -ConnectionMode $ConnectionMode `
+      -Results $Results
+    $currentDSCBlock = Get-M365DSCExportContentForResource -ResourceName $ResourceName `
+      -ConnectionMode $ConnectionMode `
+      -ModulePath $PSScriptRoot `
+      -Results $Results `
+      -GlobalAdminAccount $GlobalAdminAccount
+    $dscContent += $currentDSCBlock
+
+    Save-M365DSCPartialExport -Content $currentDSCBlock `
+      -FileName $Global:PartialExportFileName
+
+    Write-Host $Global:M365DSCEmojiGreenCheckMark
+
+
+    return $dscContent
+  }
+  catch
+  {
+    try
+    {
+      Write-Verbose -Message $_
+      $tenantIdValue = ''
+      if (-not [System.String]::IsNullOrEmpty($TenantId))
+      {
+        $tenantIdValue = $TenantId
+      }
+      elseif ($null -ne $GlobalAdminAccount)
+      {
+        $tenantIdValue = $GlobalAdminAccount.UserName.Split('@')[1]
+      }
+      Add-M365DSCEvent -Message $_ -EntryType 'Error' `
+        -EventID 1 -Source $($MyInvocation.MyCommand.Source) `
+        -TenantId $tenantIdValue
+    }
+    catch
+    {
+      Write-Verbose -Message $_
+    }
+    return ''
+  }
+}
+
+function Get-M365DSCPowerPlatformTenantSettingsJSON
+{
+  [CmdletBinding()]
+  [OutputType([System.String])]
+  param
+  (
+    [Parameter(Mandatory = $true)]
+    [System.Collections.Hashtable]
+    $Parameters
+  )
+
+  $JsonContent = @"
```

Comment:
This will not return the expected JSON string. Instead you will need to encapsulate the parameters as follow, within $()

"walkMeOptOut" : $($Parameters.WalkMeOptOut),

Reply:
Fixed in new commit